dart/lib/src/protocol/breadcrumb.dart

```diff
@@ -165,28 +170,37 @@ class Breadcrumb {
     if (data != null) {
       data = Map<String, dynamic>.from(data as Map);
     }
-
     return Breadcrumb(
       timestamp: timestamp != null ? DateTime.tryParse(timestamp) : null,
       message: json['message'],
       category: json['category'],
       data: data,
       level: levelName != null ? SentryLevel.fromName(levelName) : null,
       type: json['type'],
+      unknown: unknownFrom(json, {
+        'level',
+        'timestamp',
+        'data',
+        'message',
+        'category',
+        'type',
+      }),
```

Comment:
Defining a list (and keeping it up to date in the future) is quite error prone. Have you considered creating a wrapper for the given JSON that would keep track of consumed values & return the remainder at the end? 

Actually, why not just keep the original json map and use it when serializing?
```dart
  Map<String, dynamic> toJson() {
    return {
      ...?originalJson,
      'timestamp': formatDateAsIso8601WithMillisPrecision(timestamp),
      if (message != null) 'message': message,
      if (category != null) 'category': category,
      if (data?.isNotEmpty ?? false) 'data': data,
      if (level != null) 'level': level!.name,
      if (type != null) 'type': type,
    };
```

Reply:
My assumptions for the proposal were:
- the JSON is normally quite small
- the object already exists, so the memory is already allocated & holding a reference to it temporarily is cheaper than allocating a new object
- the time that it would be held allocated is normally short because it means it was deserialized from JSON and thus there's a good chance it's just going to get manipulated and used right away.

Reply:
Implemented an `AccessAwareMap` which uses the already allocated object and keeps track of key access to create the unknown data map.